stone/frontend/ir_generator.py

```diff
@@ -1248,3 +1259,235 @@ def mk_route_schema():
         # TODO: are we always guaranteed at least one data type?
         # pylint: disable=undefined-loop-variable
         return data_type
+
+    def _filter_namespaces_by_route_whitelist(self):
+        """
+        Given a parsed API in IR form, filter the user-defined datatypes
+        so that they include only the route datatypes and their direct dependencies.
+        """
+        assert self._routes is not None, "Missing route whitelist"
+        assert 'route_whitelist' in self._routes
+        assert 'datatype_whitelist' in self._routes
+
+        # Parse the route whitelist and populate any starting data types
+        route_data_types = []
+        for namespace_name, routes_names in self._routes['route_whitelist'].items():
+            # Error out if user supplied nonexistent namespace
+            if namespace_name not in self.api.namespaces:
+                print('Namespace %s is not defined!' % namespace_name, file=sys.stderr)
+                sys.exit(1)
```

Comment:
Not noting enough about our stone build pipeline, is there a reason this is a print + sys.exit() instead of raising an exception that is handled elsewhere.  Calling sys.exit() this deep in a process does not seem appropriate.